device.go

```diff
@@ -57,6 +57,7 @@ type DeviceResponse struct {
 	CustomLongitude                    string        `json:"custom_longitude,omitempty"`
 	CustomLatitude                     string        `json:"custom_latitude,omitempty"`
 	CreatedAt                          string        `json:"created_at,omitempty"`
+	OverallStatus                      string        `json:"overall_status,omitempty"`
```

Comment:
Great that you are looking in the details as well. I like that from a quality perspective.

The problem here is that Balena does not document all the fields that they have nor do they expose all the fields by default but they do use them. This `overall__status` field contains the actual status of the device: `offline`, `idle`, ... and it is also used on their dashboard to determine the "Status" column in the device list view.

To get this field it is necessary to use a `select` query and it will populate the field for you. In this example I would suggest to use something like:

``` bash
curl -X GET "https://api.balena-cloud.com/v6/device?\$select=id,overall_status" -H  "Content-Type: application/json"  -H "Authorization: Bearer $BALENA_API_TOKEN" -o test.json
```

Using queries like that will highly optimize the traffic from Balena to the app but it also requires more planning on what the retrieved data is going to be used for.

Reply:
Thanks for the details. How about we add a comment on the field that it needs to be explicitly select through a query parameter in order for it to be populated?